src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -523,6 +591,34 @@ public void writeHeader( final VariantContextWriter vcfWriter, final SAMSequence
         vcfWriter.writeHeader(makeVCFHeader(sequenceDictionary, defaultToolHeaderLines));
     }
 
+    /**
+     * Selects appropriate active region genotyping engine for given region
+     * @param region Current region of interest
+     * @return Active genotyping engine with correct ploidy setting for given region
+     */
+    private MinimalGenotypingEngine getLocalActiveGenotyper(Locatable region) {
+        for (final NamedFeature feature : this.ploidyRegions) {
+            if (IntervalUtils.overlaps(region, feature)) {
+                return this.ploidyToActiveEvaluationGenotyper.get(feature.getName());
+            }
+        }
+        return this.defaultActiveRegionEvaluationGenotyperEngine;
+    }
+
+    /**
+     * Selects appropriate genotyping engine for given region.
+     * @param region Current region of interest, e.g. AssemblyRegion
+     * @return Genotyping engine with correct ploidy setting for given region
+     */
+    protected HaplotypeCallerGenotypingEngine getLocalGenotypingEngine(Locatable region) {
```

Comment:
this is a somewhat ignorable comment but you could extract out a method here so these two are necessarily in lock-step and there is no risk of somebody breaking one but not the other...

Reply:
Yeah I'd like them to be covered by just one method, but unfortunately the return different types. I tried returning just the parent class and using a HashMap input that maps to the parent class, but Java doesn't seem to like this. Do you have any other suggestions, or do you think it's fine to leave as is?

Reply:
pull out a method "getPloyidyToUseAtTHISSite()" or something similar and have both of these methods call that one that has your adjudication logic and then return whatever map value is needed.